src/react-app-env.d.ts

```diff
@@ -1,26 +1,51 @@
 /// <reference types="react-scripts" />
 
-interface User {
-    id: number,
-    name: string,
-  }
-  
-  interface Post {
-    id: number,
-    userId: number,
-    title: string,
-    body: string,
-  }
-  
-  interface Comment {
-    id: number,
-    postId: number,
-    body: string,
-  }
-  
-  interface PostComment {
-    postId: number,
-    name: string,
-    email: string,
-    body: string,
-  }
\ No newline at end of file
+export interface User {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+  address: Address,
+  phone: string,
+  website: string,
+  company: Company,
+}
+
+export interface Address {
+  street: string,
+  suite: string,
+  city: string,
+  zipcode: string,
+  geo: Geo,
+}
+
+export interface Geo {
+  lat: string,
+  lng: string,
+}
+
+export interface Company {
+  name: string,
+  catchPhrase: string,
+  bs: string,
+}
+
+export interface Post {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
+}
+
+export interface Comment {
+  postId: number,
+  id: number,
+  name: string,
+  email: string,
+  body: string,
+}
+
+export interface PreparedData extends Post {
+  user?: User,
+  comments: Comment[],
+}
```

Comment:
It is not recommended to define specific interfaces and data types in the react-app-env.d.ts file, since this file is intended to define global types and variables that are used in the application. Typically, this file only defines types for global variables and objects that cannot be defined elsewhere in the application.
It's best to define interfaces and data types in separate TypeScript files within the application so they're only available in the right places. If you want to define a User interface in your application, it is recommended that you create a separate file (user.ts for example) and define the interface in that file.